add docs for styling your website to be responsive/mobile-friendly

bpas247 opened the pull request:
TODOs:
- [x] finish rough draft of each section
- [x] receive feedback
- [x] revise and finish final draft
- [x] add playable demos via [CodeSandbox](https://codesandbox.io/)

bpas247 committed:
add initial draft of topics list for docs related to responsive styling

Chris-Johnston commented:
Do you think you could also include a point about mobile-first design?

bpas247 committed:
add mobile-first design topic

bpas247 committed:
add information about viewport in the responsive css docs

bpas247 committed:
Merge branch 'add-responsive-docs' of https://github.com/UWB-ACM/Hackathon-Docs into add-responsive-docs

bpas247 committed:
add media queries section to responsive css docs

bpas247 committed:
add mobile-first design and image sections to responsive css docs

bpas247 committed:
add references to css responsive docs

bpas247 committed:
add frameworks section to css responsive docs

Chris-Johnston commented on responsive-css.md:
```diff
@@ -0,0 +1,123 @@
+# Styling your website to be responsive
+
```
Could you add a brief quip that talks about what it means to be responsive

Chris-Johnston commented on responsive-css.md:
```diff
@@ -0,0 +1,123 @@
+# Styling your website to be responsive
```
pls use title casing

Chris-Johnston commented on responsive-css.md:
````diff
@@ -0,0 +1,123 @@
+The viewport can be controlled by setting a `<meta>` tag in each web page you'd like to make responsive. The following is an example of how to use this tag:
+
+```html
+<meta name="viewport" content="width=device-width, initial-scale=1.0" />
````
might want to specify that this goes under `<head>`

Chris-Johnston commented on responsive-css.md:
```diff
@@ -0,0 +1,123 @@
+
+## Mobile-First Design
+
+Mobile-first design refers to the practice of designing your website for mobile devices before designing your website for desktop devices.
```
isn't another reason for doing this so that mobile devices load faster?

bpas247 replied:
I guess an argument could be made that a mobile-first design would improve perf by restricting the complexity of your website based on the smaller resolution of mobile devices. More info about it [here](https://sympli.io/blog/2016/07/19/3-vital-ways-a-mobile-first-approach-can-improve-web-performance/).

Chris-Johnston commented on responsive-css.md:
```diff
@@ -0,0 +1,123 @@
+The query follows the `@media` syntax, followed by the condition on which it should apply the additional CSS, followed by the additional CSS to be applied.
+
+The following example changes the background color of the body to black
+when the browser window's width is 600px or smaller.
```
Might be useful to link to where this is done on the uwbhacks website

bpas247 replied:
Ehh, if I link to a specific line on the website, there's a possibility that it might change to something else. I was planning on adding a few CodeSandbox examples to this doc before merge, that way users can see this basic example in action.

Chris-Johnston replied:
It won't change if you link to a line number of a specific commit

Chris-Johnston commented on responsive-css.md:
````diff
@@ -0,0 +1,123 @@
+
+It can be included via CDN and can be integrated into nearly any part of your webpage.
+
+```html
````
screenshots are good, I would prefer just linking to the bootstrap docs instead

bpas247 replied:
I added a CodeSandbox example, which is better than screenshots imo

Chris-Johnston replied:
While examples are nice, I still think it's no example for having an inline visual that illustrates what we are trying to show.

bpas247 replied:
I could just add the html code back in and then embed a screenshot to the output of the CodeSandbox example

Chris-Johnston commented on responsive-css.md:
```diff
@@ -0,0 +1,123 @@
+
+# References
+
+- https://www.w3schools.com/css/css_rwd_intro.asp
```
I would make these links, and not just the url

bpas247 committed:
Fix title case for title
Update viewport to describe where to put the viewport meta tag
Update references to be links rather than URL

bpas247 committed:
move bootstrap example to a live demo on CodeSandbox and add link for it

bpas247 committed:
Remove Template section
Add draft of Grid View section and raname from Grid Layout
Add a brief description of the doc

Chris-Johnston commented on responsive-css.md:
````diff
@@ -64,52 +64,7 @@ One of the most popular css frameworks to use is Bootstrap. Bootstrap combines h
-  </body>
-</html>
-```
+If you'd like to see a working Bootstrap example, click [here](https://codesandbox.io/s/vq430zj5n3)
````
I think we should stick to using the examples from the Bootstrap documentation, as they show more than just a page header. I also wonder if the codesandbox site might be a bit overwhelming for someone that we are assuming is brand new to html and css.

https://getbootstrap.com/docs/4.3/examples/

bpas247 replied:
Doesn’t seem overwhelming to me, imo. Linking to the docs should be sufficient enough though

Chris-Johnston replied:
Also consider that the first thing you see when clicking on the link is a `package.json`

bpas247 committed:
re-write bootstrap example to have embeded example image and
external link to bootstrap example docs

bpas247 committed:
Merge branch 'master' into add-responsive-docs

bpas247 committed:
remove file that resulted in a duplicate due to this PR
being merged without my knowledge

bpas247 committed:
add media queries example from uwbhacks site

bpas247 committed:
move image to proper directory

bpas247 commented:
I literally had no idea my docs for this PR was already merged, so the diffs look less impressive now :(

Chris-Johnston commented:
did @cforcomputer do that? I'm confused.

cforcomputer commented:
I was told that the docs website was being sent to professors this morning. I reorganized the master branch and took the liberty of merging the new pages with their varying levels of completion (without actually merging branches). I also placed them in what I believed to be decent directory structure. I figured updates could be made by merging the finalized branches later.
